url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/processable/GetPeriod.java

```diff
@@ -0,0 +1,98 @@
+package darwInvest.processable;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import darwInvest.environment.StrategyParameters;
+import darwin.population.Node;
+import darwin.problemObject.Processable;
+
+public class GetPeriod extends Processable<Double>{
+
+	public static final int MONTH = 0;
+	public static final int DAY_OF_MONTH = 1;
+	public static final int DAY_OF_WEEK = 2;
+	public static final int DAY_OF_WEEK_IN_MONTH = 3;
+	public static final int AM_PM = 4;
+	public static final int HOUR_OF_DAY = 5;
+	public static final int MINUTE = 6;
+	public static final int SEASON = 7;
+	public static final int FISCAL_QUARTER = 8;
+	
+	private int which;
+	
+	public GetPeriod(int which) {
+		this.which = which;
+	}
+	
+	@Override
+	public Double getValue(Object parameters, Node[] childNodes) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.